### DISCUSSION TITLE
Programatically moving elements and nesting on existing block

### DISCUSSION THREAD

**Discussion by tompkins-ct** [gaphor/gaphor#3579] [Q&A]
How would I nest items on a diagram programmatically.

I can sets position of the elements as so:
```python
if isinstance(presentation, ElementPresentation):
    presentation.handles()[NW].pos = (0.0, 0.0)
    presentation.width = item["width"]
    presentation.height = item["height"]
    presentation.matrix.set(x0=item["x"], y0=item["y"])
```

then update the diagram with

```python
diagram.update(diagram.ownedPresentation)
```

However, if the element lands inside of another it's not automatically nested and you get an un-nested result like so (class B):

![image](https://github.com/user-attachments/assets/53b953e1-5c36-48d8-9789-064b9c08290b)

Thoughts?

**Comment by mikekidner** [comment]
The presentation and the underlying model are separate.
If you are setting the position of elements on the diagram, I'd assume you're not triggering the other actions associated with dragging elements around, so the changes to the underlying model do not occur. I think you have to change the ownership of the blocks directly using the element factory.
